contracts/utils/Strings.sol

```diff
@@ -85,10 +84,46 @@ library Strings {
         return toHexString(uint256(uint160(addr)), ADDRESS_LENGTH);
     }
 
+    /**
+     * @dev Converts an `address` with fixed length of 20 bytes to its checksummed ASCII `string` hexadecimal
+     * representation, according to EIP-55.
+     */
+    function toChecksumHexString(address addr) internal pure returns (string memory) {
+        bytes memory lowercase = new bytes(40);
+        _setHexString(lowercase, 0, uint160(addr));
```

Comment:
We have the `uint160(addr)` twice. This should be better for both readability and perhaps saving one `PUSH` here and there.

```suggestion
        uint160 addrValue = uint160(address)
        _setHexString(lowercase, 0, addrValue);
```

Reply:
good catch